internal/versions/versions.go

```diff
@@ -206,37 +206,30 @@ func (h *handler) ValidateReleaseImageForRHCOS(rhcosVersion, cpuArchitecture str
 	return errors.Errorf("The requested RHCOS version (%s, arch: %s) does not have a matching OpenShift release image", rhcosVersion, cpuArchitecture)
 }
 
-func (h *handler) AddReleaseImage(releaseImageUrl, pullSecret, ocpReleaseVersion string, cpuArchitectures []string) (*models.ReleaseImage, error) {
-	var err error
-	var cpuArchitecture string
-	var osImage *models.OsImage
-
-	// If release version or cpu architectures are not specified, use oc to fetch values.
-	// If cpu architecture is passed as "multiarch" instead of unwrapped architectures, recalculate it.
-	if ocpReleaseVersion == "" || len(cpuArchitectures) == 0 || cpuArchitectures[0] == common.MultiCPUArchitecture {
-		// Get openshift version from release image metadata (oc adm release info)
-		ocpReleaseVersion, err = h.releaseHandler.GetOpenshiftVersion(h.log, releaseImageUrl, "", pullSecret)
-		if err != nil {
-			return nil, err
-		}
-		h.log.Debugf("For release image %s detected version: %s", releaseImageUrl, ocpReleaseVersion)
+func (h *handler) AddReleaseImage(releaseImageUrl, pullSecret string) (*models.ReleaseImage, error) {
+	// Get openshift version from release image metadata (oc adm release info)
+	ocpReleaseVersion, err := h.releaseHandler.GetOpenshiftVersion(h.log, releaseImageUrl, "", pullSecret)
+	if err != nil {
+		return nil, err
+	}
+	h.log.Debugf("For release image %s detected version: %s", releaseImageUrl, ocpReleaseVersion)
 
-		// Get CPU architecture from release image. For single-arch image the returned list will contain
-		// as single entry with the architecture. For multi-arch image the list will contain all the architectures
-		// that the image references to.
-		cpuArchitectures, err = h.releaseHandler.GetReleaseArchitecture(h.log, releaseImageUrl, "", pullSecret)
-		if err != nil {
-			return nil, err
-		}
-		h.log.Debugf("For release image %s detected architecture: %s", releaseImageUrl, cpuArchitectures)
+	// Get CPU architecture from release image. For single-arch image the returned list will contain
+	// as single entry with the architecture. For multi-arch image the list will contain all the architectures
+	// that the image references to.
+	cpuArchitectures, err := h.releaseHandler.GetReleaseArchitecture(h.log, releaseImageUrl, "", pullSecret)
```

Comment:
So shouldn't we now check if the releaseImageUrl exists in cache first?
I.e. To avoid calling GetReleaseArchitecture every time, as this func is not very optimized (might execute two calls, for both skopeo and oc)